src/lib/support/StringBuilder.h

```diff
@@ -89,13 +91,27 @@ class StringBuilderBase
     }
 };
 
-/// a preallocated sized string builder
-template <size_t kSize>
+/// A preallocated sized string builder
+/// Default buffer size is 257
+template <size_t kSize = 257>
 class StringBuilder : public StringBuilderBase
 {
 public:
+    /// Default constructor
     StringBuilder() : StringBuilderBase(mBuffer, kSize) {}
 
+    /// Constructor for char * and length
+    StringBuilder(const char * data, size_t length) : StringBuilder() { AddFormat("%.*s", static_cast<int>(length), data); }
+
+    /// Constructor for uint8_t * and length
+    StringBuilder(const uint8_t * data, size_t length) : StringBuilder(Uint8::to_const_char(data), length) {}
+
+    /// Constructor for CharSpan
+    StringBuilder(const CharSpan & span) : StringBuilder(span.data(), span.size()) {}
+
+    /// Constructor for ByteSpan
+    StringBuilder(const ByteSpan & span) : StringBuilder(Uint8::to_const_char(span.data()), span.size()) {}
```

Comment:
Again, the casting needs to be at callsites, so the lossy conversion is not hidden.